File: Backend/LogisticsPlatform.Application/DTOs/Customers/Addresses/CreateCustomerAddressDto.cs
using LogisticsPlatform.Domain.Enums;

public class CreateCustomerAddressDto
{
    public Guid CustomerId { get; set; }
    public string AddressLine1 { get; set; } = string.Empty;
    public string? AddressLine2 { get; set; }
    public string City { get; set; } = string.Empty;
    public string? State { get; set; }
    public string Country { get; set; } = string.Empty;
    public string? PostalCode { get; set; }
    public CustomerAddressType Type { get; set; }
    public bool IsPrimary { get; set; }
    public bool isActive { get; set; }
}

File: Frontend/src/app/core/models/customers/addresses/create-customer-address.dto.ts
import { CustomerAddressType } from "./customer-address-type";

export interface CreateCustomerAddressDto {
    customerId?: string;
    addressLine1: string;
    addressLine2?: string;
    city: string;
    state?: string;
    country: string;
    postalCode?: string;
    type: CustomerAddressType;
    isPrimary: boolean;
    isActive: boolean;
}
